vpr/src/route/route_common.cpp

```diff
@@ -713,24 +690,49 @@ float get_rr_cong_cost(int inode) {
                     continue; //Already included above
                 }
 
-                cost += get_single_rr_cong_cost(node);
+                cost += get_single_rr_cong_cost(node, pres_fac);
             }
         }
     }
     return (cost);
 }
 
-/* Returns the congestion cost of using this rr_node, *ignoring*
- * non-configurable edges */
-static float get_single_rr_cong_cost(int inode) {
+/* Returns the congestion cost of using this rr_node,
+ * *ignoring* non-configurable edges */
+float get_single_rr_cong_cost(int inode, float pres_fac) {
```

Comment:
This is a very hot function that is calling 3 other short functions.  Check to be sure it is inlined (can look at asm, or use the debugger on the optimized code).  Could put inline on the functions to give a hint. 
If they aren't inlined, I would also measure the speed of manually inlining the function bodies (simple) of the 3 short functions called (or at least the very simple base and acc_cost ones).  With code this hot and functions this short, the function call overhead can be significant.